File: web/src/api/smartStorageDoor.js
import service from '@/utils/request'

// @Tags SmartStorageDoor
// @Summary 创建SmartStorageDoor
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageDoor true "创建SmartStorageDoor"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /smartStorageDoor/createSmartStorageDoor [post]
export const createSmartStorageDoor = (data) => {
     return service({
         url: "/smartStorageDoor/createSmartStorageDoor",
         method: 'post',
         data
     })
 }


// @Tags SmartStorageDoor
// @Summary 删除SmartStorageDoor
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageDoor true "删除SmartStorageDoor"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /smartStorageDoor/deleteSmartStorageDoor [delete]
 export const deleteSmartStorageDoor = (data) => {
     return service({
         url: "/smartStorageDoor/deleteSmartStorageDoor",
         method: 'delete',
         data
     })
 }

// @Tags SmartStorageDoor
// @Summary 删除SmartStorageDoor
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除SmartStorageDoor"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /smartStorageDoor/deleteSmartStorageDoor [delete]
 export const deleteSmartStorageDoorByIds = (data) => {
     return service({
         url: "/smartStorageDoor/deleteSmartStorageDoorByIds",
         method: 'delete',
         data
     })
 }

// @Tags SmartStorageDoor
// @Summary 更新SmartStorageDoor
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageDoor true "更新SmartStorageDoor"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /smartStorageDoor/updateSmartStorageDoor [put]
 export const updateSmartStorageDoor = (data) => {
     return service({
         url: "/smartStorageDoor/updateSmartStorageDoor",
         method: 'put',
         data
     })
 }


// @Tags SmartStorageDoor
// @Summary 用id查询SmartStorageDoor
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageDoor true "用id查询SmartStorageDoor"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /smartStorageDoor/findSmartStorageDoor [get]
 export const findSmartStorageDoor = (params) => {
     return service({
         url: "/smartStorageDoor/findSmartStorageDoor",
         method: 'get',
         params
     })
 }


// @Tags SmartStorageDoor
// @Summary 分页获取SmartStorageDoor列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取SmartStorageDoor列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /smartStorageDoor/getSmartStorageDoorList [get]
 export const getSmartStorageDoorList = (params) => {
     return service({
         url: "/smartStorageDoor/getSmartStorageDoorList",
         method: 'get',
         params
     })
 }

File: server/router/smartStorageDoor.go
package router

import (
	"gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitSmartStorageDoorRouter(Router *gin.RouterGroup) {
	SmartStorageDoorRouter := Router.Group("smartStorageDoor").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		SmartStorageDoorRouter.POST("createSmartStorageDoor", v1.CreateSmartStorageDoor)   // 新建SmartStorageDoor
		SmartStorageDoorRouter.DELETE("deleteSmartStorageDoor", v1.DeleteSmartStorageDoor) // 删除SmartStorageDoor
		SmartStorageDoorRouter.DELETE("deleteSmartStorageDoorByIds", v1.DeleteSmartStorageDoorByIds) // 批量删除SmartStorageDoor
		SmartStorageDoorRouter.PUT("updateSmartStorageDoor", v1.UpdateSmartStorageDoor)    // 更新SmartStorageDoor
		SmartStorageDoorRouter.GET("findSmartStorageDoor", v1.FindSmartStorageDoor)        // 根据ID获取SmartStorageDoor
		SmartStorageDoorRouter.GET("getSmartStorageDoorList", v1.GetSmartStorageDoorList)  // 获取SmartStorageDoor列表
	}
}
